purchases/src/main/kotlin/com/revenuecat/purchases/ListenerConversionsCommon.kt

```diff
@@ -0,0 +1,150 @@
+package com.revenuecat.purchases
+
+import com.revenuecat.purchases.interfaces.GetStoreProductsCallback
+import com.revenuecat.purchases.interfaces.ProductChangeCallback
+import com.revenuecat.purchases.interfaces.PurchaseCallback
+import com.revenuecat.purchases.interfaces.ReceiveCustomerInfoCallback
+import com.revenuecat.purchases.interfaces.ReceiveOfferingsCallback
+import com.revenuecat.purchases.interfaces.SyncPurchasesCallback
+import com.revenuecat.purchases.models.StoreProduct
+import com.revenuecat.purchases.models.StoreTransaction
+
+internal val ON_ERROR_STUB: (error: PurchasesError) -> Unit = {}
+internal val ON_PURCHASE_ERROR_STUB: (error: PurchasesError, userCancelled: Boolean) -> Unit = { _, _ -> }
+
+internal fun purchaseCompletedCallback(
+    onSuccess: (purchase: StoreTransaction, customerInfo: CustomerInfo) -> Unit,
+    onError: (error: PurchasesError, userCancelled: Boolean) -> Unit,
+) = object : PurchaseCallback {
+    override fun onCompleted(storeTransaction: StoreTransaction, customerInfo: CustomerInfo) {
+        onSuccess(storeTransaction, customerInfo)
+    }
+
+    override fun onError(error: PurchasesError, userCancelled: Boolean) {
+        onError(error, userCancelled)
+    }
+}
+
+internal fun productChangeCompletedListener(
+    onSuccess: (purchase: StoreTransaction?, customerInfo: CustomerInfo) -> Unit,
+    onError: (error: PurchasesError, userCancelled: Boolean) -> Unit,
+) = object : ProductChangeCallback {
+    override fun onCompleted(purchase: StoreTransaction?, customerInfo: CustomerInfo) {
+        onSuccess(purchase, customerInfo)
+    }
+
+    override fun onError(error: PurchasesError, userCancelled: Boolean) {
+        onError(error, userCancelled)
+    }
+}
+
+internal fun getStoreProductsCallback(
+    onReceived: (storeProducts: List<StoreProduct>) -> Unit,
+    onError: (error: PurchasesError) -> Unit,
+) = object : GetStoreProductsCallback {
+    override fun onReceived(storeProducts: List<StoreProduct>) {
+        onReceived(storeProducts)
+    }
+
+    override fun onError(error: PurchasesError) {
+        onError(error)
+    }
+}
+
+internal fun receiveOfferingsCallback(
+    onSuccess: (offerings: Offerings) -> Unit,
+    onError: (error: PurchasesError) -> Unit,
+) = object : ReceiveOfferingsCallback {
+    override fun onReceived(offerings: Offerings) {
+        onSuccess(offerings)
+    }
+
+    override fun onError(error: PurchasesError) {
+        onError(error)
+    }
+}
+
+internal fun receiveCustomerInfoCallback(
+    onSuccess: (customerInfo: CustomerInfo) -> Unit?,
+    onError: (error: PurchasesError) -> Unit?,
+) = object : ReceiveCustomerInfoCallback {
+    override fun onReceived(customerInfo: CustomerInfo) {
+        onSuccess?.invoke(customerInfo)
+    }
+
+    override fun onError(error: PurchasesError) {
+        onError?.invoke(error)
+    }
+}
+
+internal fun syncPurchasesListener(
```

Comment:
I think this can be moved to listenerConversions in the defaults flavor

Reply:
Moved in https://github.com/RevenueCat/purchases-android/pull/1153/commits/fcbab478b4d86b669e944a7e179674be17f36b92